pkg/auth/group/group_adder.go

```diff
@@ -0,0 +1,32 @@
+package group
+
+import (
+	"net/http"
+
+	"github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/auth/authenticator"
+)
+
+// GroupAdder wraps a request authenticator, and adds the specified groups to the returned user when authentication succeeds
+type GroupAdder struct {
+	Authenticator authenticator.Request
+	Groups        []string
+}
+
+func (g *GroupAdder) AuthenticateRequest(req *http.Request) (api.UserInfo, bool, error) {
```

Comment:
for now, going to claim YAGNI... request authenticators are the root of all other authenticator usage for the API... when I need to lookup/add group info at some other point, I'll rethink this